cmd/unset.go

```diff
@@ -35,7 +35,20 @@ func unsetCmd(ctx *cli.Context) error {
 		return errs.NewUsageExitError(msg, ctx)
 	}
 
-	cred, err := setCredential(ctx, args[0], func() *apitypes.CredentialValue {
+	pathexp, _, err := determineCredential(ctx, args[0])
+	if err != nil {
+		return errs.NewErrorExitError("Could not unset credential", err)
+	}
+
+	preamble := fmt.Sprintf("You are about to unset \"%s\". This cannot be undone.", pathexp.String())
+
+	abort := ConfirmDialogue(ctx, nil, &preamble)
```

Comment:
`bool, err` won't work because the abstraction is meant to construct the exit error.